### PR TITLE
KSPACE-20: Drop the distinction between host & member ToolchainClusters

### PR DESCRIPTION
**Pull request by fbm3307**
This is related to Drop the distinction between host & member ToolchainClusters
Related PRs:
Toolchain-Common - https://github.com/codeready-toolchain/toolchain-common/pull/359
Host-Operator - https://github.com/codeready-toolchain/host-operator/pull/971
Member-Operator - https://github.com/codeready-toolchain/member-operator/pull/530
Registration-Service - https://github.com/codeready-toolchain/registration-service/pull/402
Sandbox-sre - https://github.com/codeready-toolchain/sandbox-sre/pull/1524

### CODE REVIEW

File: test/e2e/toolchaincluster_test.go [hunk 1]
```diff
@@ -117,48 +109,26 @@ func verifyToolchainCluster(t *testing.T, await *wait.Awaitility, otherAwait *wa
...
 				"namespace": await.Namespace,
-				"type":      string(await.Type),
 			}), wait.UntilToolchainClusterHasCondition(*wait.ReadyToolchainCluster),
-			toolchainClusterWaitCriterionBasedOnType(await.Type),
 		)
 		require.NoError(t, err)
 	})
 }
 
-// toolchainClusterWaitCriterionBasedOnType returns ToolchainClusterWaitCriterion based on toolchaincluster type (member or host)
```
[the remark this answers is not in the document]
Comment (fbm3307) [reply test/e2e/toolchaincluster_test.go:145]: okay right, thanks for pointing this out

[the remark this answers is not in the document]
Comment (fbm3307) [reply test/e2e/toolchaincluster_test.go:145]: updated it

File: test/e2e/toolchaincluster_test.go [hunk 2]
```diff
@@ -30,15 +29,14 @@ func TestToolchainClusterE2E(t *testing.T) {
 // in the target cluster type operator
 func verifyToolchainCluster(t *testing.T, await *wait.Awaitility, otherAwait *wait.Awaitility) {
 	// given
-	current, ok, err := await.GetToolchainCluster(t, otherAwait.Type, otherAwait.Namespace, nil)
+	current, ok, err := await.GetToolchainCluster(t, otherAwait.Namespace, nil)
 	require.NoError(t, err)
 	require.True(t, ok, "ToolchainCluster should exist")
 
-	t.Run("create new ToolchainCluster with correct data and expect to be ready for cluster type "+string(await.Type), func(t *testing.T) {
+	t.Run("create new ToolchainCluster with correct data and expect to be ready for cluster", func(t *testing.T) {
```
Comment (MatousJobanek) [inline test/e2e/toolchaincluster_test.go:36]: let's somehow distinguish between the two execution of this function:
  ```suggestion
	t.Run(fmt.Sprintf("create new ToolchainCluster based on '%s' with correct data and expect to be ready", await.ClusterName), func(t *testing.T) {
```

File: test/e2e/toolchaincluster_test.go [hunk 3]
```diff
@@ -62,35 +60,29 @@ func verifyToolchainCluster(t *testing.T, await *wait.Awaitility, otherAwait *wa
...
 				"namespace": await.Namespace,
-				"type":      string(await.Type),
 			}), wait.UntilToolchainClusterHasCondition(*wait.ReadyToolchainCluster),
-			toolchainClusterWaitCriterionBasedOnType(await.Type),
 		)
 		require.NoError(t, err)
 	})
 
-	t.Run("create new ToolchainCluster with incorrect data and expect to be offline for cluster type "+string(await.Type), func(t *testing.T) {
+	t.Run("create new ToolchainCluster with incorrect data and expect to be offline for cluster type ", func(t *testing.T) {
```
Comment (MatousJobanek) [inline test/e2e/toolchaincluster_test.go:82]:
```suggestion
	t.Run(fmt.Sprintf("create new ToolchainCluster based on '%s' with incorrect data and expect to be offline", await.ClusterName), func(t *testing.T) {
```

File: test/e2e/toolchaincluster_test.go [hunk 4]
```diff
@@ -30,15 +29,14 @@ func TestToolchainClusterE2E(t *testing.T) {
...
-	current, ok, err := await.GetToolchainCluster(t, otherAwait.Type, otherAwait.Namespace, nil)
+	current, ok, err := await.GetToolchainCluster(t, otherAwait.Namespace, nil)
 	require.NoError(t, err)
 	require.True(t, ok, "ToolchainCluster should exist")
 
-	t.Run("create new ToolchainCluster with correct data and expect to be ready for cluster type "+string(await.Type), func(t *testing.T) {
+	t.Run("create new ToolchainCluster with correct data and expect to be ready for cluster", func(t *testing.T) {
 		// given
-		name := "new-ready-" + string(otherAwait.Type)
+		name := "new-ready"
```
Comment (MatousJobanek) [inline test/e2e/toolchaincluster_test.go:38]: let's make sure that the toolchaincluster won't conflict in the two following runs:
```suggestion
		name := "new-ready-" + await.ClusterName
```

File: test/e2e/toolchaincluster_test.go [hunk 5]
```diff
@@ -62,35 +60,29 @@ func verifyToolchainCluster(t *testing.T, await *wait.Awaitility, otherAwait *wa
...
-			toolchainClusterWaitCriterionBasedOnType(await.Type),
 		)
 		require.NoError(t, err)
 	})
 
-	t.Run("create new ToolchainCluster with incorrect data and expect to be offline for cluster type "+string(await.Type), func(t *testing.T) {
+	t.Run("create new ToolchainCluster with incorrect data and expect to be offline for cluster type ", func(t *testing.T) {
 		// given
-		name := "new-offline-" + string(otherAwait.Type)
+		name := "new-offline"
```
Comment (MatousJobanek) [inline test/e2e/toolchaincluster_test.go:84]:
```suggestion
		name := "new-offline-" + await.ClusterName
```

**Review by MatousJobanek — COMMENTED** [review]
Looks good, just a small proposal for better reading and to avoid conflicts

File: test/e2e/toolchaincluster_test.go (hunk 2 above)
Comment (fbm3307) [reply test/e2e/toolchaincluster_test.go]: done

File: test/e2e/toolchaincluster_test.go (hunk 4 above)
Comment (fbm3307) [reply test/e2e/toolchaincluster_test.go]: updated it

**Review by MatousJobanek — APPROVED** [review]
Thanks :+1:

**Comment by fbm3307** [comment]
looks some flaky test failure 
/test e2e

**Pull request merged**